cibuildwheel/macos.py

```diff
@@ -232,8 +231,26 @@ def setup_python(
         )
         sys.exit(1)
 
-    # install pip & wheel
-    call(["python", get_pip_script, *dependency_constraint_flags], env=env, cwd="/tmp")
+    # ensure pip is installed
+    call(["python", "-m", "ensurepip"], env=env, cwd="/tmp")
+
+    # upgrade to the version matching our constraints
+    # if necessary, reinstall it to ensure that it's installed as 'pip'
+    requires_reinstall = not (installation_bin_path / "pip").exists()
```

Comment:
maybe move `python -m ensurepip` below `requires_reinstall` to get rid of this call as well ?

```python
requires_reinstall = not (installation_bin_path / "pip").exists()
if requires_reinstall:
    # ensure pip is installed
    call(["python", "-m", "ensurepip"], env=env, cwd="/tmp")
```

Reply:
hmm. yeah, I guess it would be a little faster? I thought 'ensurepip' was a quick noop when it's installed, but it still takes around a second on my machine. I'll try your suggestion.